src/globalvars.cpp

```diff
@@ -258,19 +303,99 @@ void GlobalVars::verify_maps_found(
     const std::unordered_map<std::string, struct bpf_map *>
         &section_name_to_global_vars_map)
 {
-  for (const auto &[name, config] : global_var_map_) {
+  for (const auto &[name, config] : added_global_vars_) {
     if (!section_name_to_global_vars_map.contains(config.section)) {
       LOG(BUG) << "No map found in " << config.section
                << " which is needed to set global variable " << name;
     }
   }
 }
 
+Result<GlobalVarVals> GlobalVars::get_named_param_vals(
+    std::unordered_map<std::string, std::string> named_params)
+{
+  GlobalVarVals named_param_vals;
+
+  std::string expected_args_str;
+  for (const auto &[name, default_val] : named_param_defaults_) {
+    auto &config = added_global_vars_.at(name);
+    auto it = named_params.find(name);
+    std::string val = default_val;
+
+    if (it != named_params.end()) {
+      val = it->second;
+      named_params.erase(name);
+    }
+
+    if (config.type == GlobalVarType::integer) {
+      try {
+        named_param_vals[name] = std::stoll(val);
+      } catch (const std::invalid_argument &) {
+        return make_error<NamedParamError>(
+            "Command line option --" + name +
+            " is expecting an integer. Got: " + val);
+      } catch (const std::out_of_range &) {
+        return make_error<NamedParamError>("Value for command line option --" +
+                                           name +
+                                           " is out of range. Got: " + val);
+      }
+    } else if (config.type == GlobalVarType::boolean) {
+      if (val == "--") {
```

Comment:
Huh? This is `--foo=--` ? Why is this not an error?

Reply:
Oh, I see that this is "TRUTH_VALUE". I think it is actually a presence value.

Why do you need this? Can't presence in the map be an indicator of presence (e.g. just make `--foo` and `--foo=` equivalent in terms of representation).